### ISSUE TITLE
Build

### ISSUE THREAD

**Issue by ViPErCZ** [hotstreams/limitless-engine#24]
Please help, 
I wanted to try the app, but I can't build > 

[100%] Linking C executable ../../../../bin/glewinfo
/usr/bin/ld: CMakeFiles/glewinfo.dir/__/__/src/glewinfo.c.o: in function `glewCreateContext':
glewinfo.c:(.text+0x32209): undefined reference to `glXQueryExtension'

demo build... but not execute

first error
tesselation.tcs: 0:608(38): error: no matching function for call to `getVertexNormal(int)'; candidates are:
0:608(38): error:    vec3 getVertexNormal(uint)

if i fix then,

error: Input block `_vertex_data' is not an output of the previous stage

Thanks.

**Comment by ViPErCZ** [comment]
And today version has new error... 
limitless-engine/include/limitless/instances/instanced_instance.hpp:83:28: error: conflicting return type specified for ‘virtual Limitless::InstancedInstance<Limitless::ModelInstance>* Limitless::InstancedInstance<Limitless::ModelInstance>::clone()’
   83 |         InstancedInstance* clone() noexcept override {

and tess_control_input.glsl has wrong on runtime error
tesselation.tcs: 0:621(38): error: no matching function for call to `getVertexNormal(int)'; candidates are:
0:621(38): error:    vec3 getVertexNormal(uint)

**Comment by hotstreams** [comment]
Yeah, Sorry for that, definetely my bad.

>tesselation.tcs: 0:621(38): error: no matching function for call to `getVertexNormal(int)'; candidates are:
0:621(38): error: vec3 getVertexNormal(uint)

There was something about implicit casts from int to uint, should be fixed now at [f858540](https://github.com/hotstreams/limitless-engine/commit/f8585407b4d2f944e1e3c060353ac3832ffce52a)

**Comment by ViPErCZ** [comment]
Not completed.... getVertexNormal must be to int type...  by call
_out_data[gl_InvocationID].normal = getVertexNormal(gl_InvocationID);

gl_InvocationID is int type

terminate called after throwing an instance of 'Limitless::ms::material_compilation_error'
  what():  tesselation_samplefailed to compile /home/viper/Plocha/limitless-engine/assets/../shaders/tesselation/tesselation.tcs: 0:621(38): error: no matching function for call to `getVertexNormal(int)'; candidates are:
0:621(38): error:    vec3 getVertexNormal(uint)
0:624(46): error: no matching function for call to `getVertexPosition(int)'; candidates are:
0:624(46): error:    vec3 getVertexPosition(uint)
0:625(34): error: no matching function for call to `getVertexUV(int)'; candidates are:
0:625(34): error:    vec2 getVertexUV(uint)

**Comment by Lee-Marry** [comment]
@ViPErCZ Hi, Can you run the project sucessfully?I find the demo does not include files "Textures"

**Comment by Lee-Marry** [comment]
for example
![image](https://github.com/hotstreams/limitless-engine/assets/73864469/dd670398-ef1c-45ad-9c17-51bc998d3bfb)
![image](https://github.com/hotstreams/limitless-engine/assets/73864469/9c2e5f57-8c10-4cb7-a65d-7d687635137b)

**Comment by ViPErCZ** [comment]
https://github.com/hotstreams/limitless-engine/commit/78b6d5a29158e527efe5d990dd8acf67f587fdd2

assets removed. I have old function commit e24fe45cf55a834b7bfd6283f796a5883df015a4
I have no head checkout. This is maybe bug. ;-)

**Comment by Lee-Marry** [comment]
Thanks for your reply ,I have found from https://github.com/Tehsapper/graphics-engine

**Comment by ViPErCZ** [comment]
Yes, using fork is solution... ;-)

**Comment by ViPErCZ** [comment]
yes... uint not match int in argument this getVertexUV func... you must fix... ideal when this bug fix owner this repository

**Comment by Lee-Marry** [comment]
I took a shortcut and deleted the tcs file,I built it sucessfully finally,and run the project normally

**Comment by Lee-Marry** [comment]
The video I take is https://www.bilibili.com/video/BV1wF4m1g7Rf/?spm_id_from=333.999.0.0&vd_source=c190caf289a70529fcd37e9d7dadedaf

**Comment by Lee-Marry** [comment]
I want change it from glew to glad,and create a special effects library based on this
